src/cpp/shared/adventure/adv.cpp

```diff
@@ -341,6 +374,248 @@ void advManager::ComputerMonsterInteract(mapCell *cell, hero *computer, int *a1)
 	this->ComputerMonsterInteract_orig(cell, computer, a1);
 }
 
+void advManager::UpdateTownLocators(int a2, int updateScreen) { 
```

Comment:
I've told you what needs to be done to the GUI code.

Reply:
There is no defined constant for type 8 GUI_MESSAGE_CODE so I left this exact part intact